controllers/virtualmachine/virtualmachine_controller.go

```diff
@@ -87,12 +89,31 @@ func AddToManager(ctx *context.ControllerManagerContext, mgr manager.Manager) er
 			handler.EnqueueRequestsFromMapFunc(csBindingToVMMapperFn(ctx, r.Client)))
 	}
 
-	if !lib.IsNamespacedVMClassFSSEnabled() {
-		builder = builder.Watches(&source.Kind{Type: &vmopv1.VirtualMachineClassBinding{}},
-			handler.EnqueueRequestsFromMapFunc(classBindingToVMMapperFn(ctx, r.Client)))
+	// Predicate functions to determine when a reconcile should be triggered
+	// Used to only watch for VMClass Create and Delete event
+	eventFilter := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
```

Comment:
Hmm.  IIRC, you only need to specify the functions that you want to "change".  IOW, just the Update function here.  I don't know what GenericFunc is used for.  Maybe worth checking?

- https://sdk.operatorframework.io/docs/building-operators/golang/references/event-filtering/#using-predicates
- https://stuartleeks.com/posts/kubebuilder-event-filters-part-2-update/#implementing-the-filter
-